nautobot/extras/plugins/__init__.py

```diff
@@ -358,6 +383,39 @@ def register_jobs(class_list):
     # That is done in response to the `nautobot_database_ready` signal, see nautobot.extras.signals.refresh_job_models
 
 
+class PluginFilterExtension:
+    """Class that may be returned by a registered Filter Extension function."""
+
+    model = None
+
+    def filterset(self):
+        return None
+
+    def filter_form(self):
+        return None
+
+
+def register_filter_extensions(class_list, name):
+    """
+    Register a list of PluginFilterExtension classes
```

Comment:
Please document the new registry key(s) in `nautobot/docs/development/application-registry.md` as well.

Reply:
TODO